scripts/import-users.py

```diff
@@ -0,0 +1,410 @@
+#!/usr/bin/env python3
+
+import os
+import re
+import base64
+import argparse
+import warnings
+from pathlib import Path
+from getpass import getpass
+
+import requests
+import pandas as pd
+
+
+def main():
+    server, data_directory = CommandLineParser().parse()
+    api = API(server)
+    cohorts = ProjectDataParser().parse(data_directory)
+    ProjectProcessor(api).parse(cohorts)
+
+
+class ProjectDataParser:
+    def parse(self, directory):
+        """
+        Method to parse all TSVs in a user supplied directory, assuming all those TSVs are export
+        TSVs from the export script. Returns a dictionary with a (parsed and sanitized) cohort name
+        as key and it's user data as value.
+
+        Args:
+            directory:
+                Absolute path to the directory containing the user cohort export TSVs.
+
+        Returns:
+            dict:
+                Returns a dictionary containing (key) cohort name, parsed to adhere to AWS
+                bucket standards and (value) cohort user data.
+        """
+        return_dict = {}
+        files = os.listdir(directory)
+        for file in files:
+            if not file.endswith((".tsv", ".tsv.gz")):
+                continue
+            full_path = os.path.join(directory, file)
+            project_name = self._parse_filename(file)
+            return_dict[project_name] = self._load_tsv(full_path)
+        return return_dict
+
+    @staticmethod
+    def _parse_filename(filename):
+        """
+        Function to parse the filename to the project name.
+
+        Args:
+            filename:
+                The string of the filename of the TSV.
+        Returns:
+            string:
+                Returns a string that is AWS bucket naming compliant.
+        """
+        # First, break off file extension
+        filename = filename.replace(".tsv.gz", "")
+        filename = filename.replace(".tsv", "")
+        # Then create an intermediate with only valid characters
+        name = "".join(
+            [c for c in filename.replace("-", "") if c.isalnum() or c == "_"]
+        )
+        # Then clean up starting spaces
+        name = re.sub(r"^_*", "", name)
+        name = re.sub(r"_+", "-", name)
```

Comment:
You can replace these with `.strip()` after breaking file extension. https://docs.python.org/3/library/stdtypes.html?highlight=strip#str.strip

Reply:
Spaces was incorrect in the comment line, it's to strip starting underscores. The second regex is to replace 1 or more underscores with a single - line that is allowed in AWS bucket naming scheme.